store/rootmulti/store.go

```diff
@@ -167,7 +166,32 @@ func (rs *Store) CacheMultiStore() types.CacheMultiStore {
 // Implements interface MultiStore
 // used to createQueryContext, abci_query or grpc query service.
 func (rs *Store) CacheMultiStoreWithVersion(version int64) (types.CacheMultiStore, error) {
-	panic("rootmulti Store don't support historical query service")
+	if version == 0 || (rs.lastCommitInfo != nil && version == rs.lastCommitInfo.Version) {
```

Comment:
not sure if need consider sdk46Compact, otherwise LGTM

Reply:
no, that is only about app hash calculation.